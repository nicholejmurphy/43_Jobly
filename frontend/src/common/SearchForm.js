import React, { useState } from "react";
import { Form, Row, Col, Input, Button } from "reactstrap";

/** Search input for { CompanyList & JobsList }
 *  - api call is passed into form via "searchFor"
 *    prop from parent component
 *  - Api call occurs in parent component, "searchFor"
 *    only just triggered here
 *
 */
function SearchForm({ search }) {
  const [searchTerm, setSearchTerm] = useState("");

  // Updates serchterm state on form input change
  function handleChange(e) {
    setSearchTerm(e.target.value);
  }

  // On submit, serach() is triggered, passing
  // the search term back to parent.
  //  - Trimming extra spaces to keep from passing
  //    those as part of the filtering term
  async function handleSubmit(e) {
    e.preventDefault();
    search(searchTerm.trim() || undefined);
    setSearchTerm(searchTerm.trim());
  }

  return (
    <div className="SearchForm mb-4 mt-4 ml-3 vw-100">
      <Form onSubmit={handleSubmit}>
        <Row>
          <Col>
            <Input
              className="shadow"
              id="search"
              name="search"
              type="text"
              placeholder="Enter search term..."
              value={searchTerm}
              onChange={handleChange}
            />
          </Col>
          <Col>
            {" "}
            <Button className="bg-info shadow">Search</Button>
          </Col>
        </Row>
      </Form>
    </div>
  );
}

export default SearchForm;
