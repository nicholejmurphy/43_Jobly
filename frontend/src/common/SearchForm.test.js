import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import SearchForm from "./SearchForm";
import { MemoryRouter } from "react-router";

// smoke test
it("renders without crashing", function () {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <SearchForm />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <SearchForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
