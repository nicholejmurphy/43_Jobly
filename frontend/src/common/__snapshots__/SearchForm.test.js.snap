// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SearchForm mb-4 mt-4 ml-3 vw-100"
  >
    <form
      class=""
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <input
            class="form-control"
            id="search"
            name="search"
            placeholder="Enter search term..."
            type="text"
            value=""
          />
        </div>
        <div
          class="col"
        >
           
          <button
            class="bg-warning btn btn-secondary"
          >
            Search
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
